workspaces/topology/plugins/topology/src/components/Topology/TopologyToolbar.tsx

```diff
@@ -69,23 +70,17 @@ const TopologyToolbar = ({ showFilters }: TopologyToolbarProps) => {
   };
 
   const onDisplayOptionChange = (
-    e: React.ChangeEvent | React.MouseEvent,
-    selection: string | SelectOptionObject,
+    e: React.MouseEvent | undefined,
+    selection: string | number | undefined,
   ) => {
-    if (filters && filters.length !== 0) {
-      const index = filters?.findIndex(f => f.id === selection);
-
-      if (index !== undefined && index > -1) {
-        const filter = {
-          ...filters[index],
-          value: (e.target as HTMLInputElement).checked,
-        };
-        setAppliedTopologyFilters?.([
-          ...filters.slice(0, index),
-          filter,
-          ...filters.slice(index + 1),
-        ] as TopologyDisplayOption[]);
-      }
+    if (filters && filters.length !== 0 && selection && e) {
+      const index = filters.findIndex(filter => filter.value === selection);
+      const newFilters = [...filters];
+      newFilters[index] = {
+        ...newFilters[index],
+        isSelected: !newFilters[index].isSelected,
+      };
+      setAppliedTopologyFilters?.(newFilters as TopologyDisplayOption[]);
```

Comment:
Is this type assertion necessary?

Reply:
it's no longer needed, thanks for catching :p